elixir/apps/web/test/web/live/sign_up_test.exs

```diff
@@ -68,6 +68,84 @@ defmodule Web.Live.SignUpTest do
     end)
   end
 
+  test "allows override to create new account and send a welcome email", %{conn: conn} do
+    Domain.Config.put_env_override(:outbound_email_adapter_configured?, true)
+    Domain.Config.feature_flag_override(:sign_up, false)
```

Comment:
I think if the flag is `false` we should not allow sign-ups at all, otherwise the configuration will get confusing. Maybe it should be like this:

flag | domains | results
true | []            | allowed from any domain (no whitelisting enabled)
true | [..]          | only allowed from specified domains
false | ...           | sign up is completely disabled

Reply:
Alternatively we can remove `sign_up_enabled` and just rely on domains list value, so if it's nil (env variable is not set) - all sign ups enabled, if it's empty - none.

Reply:
😄 this was literally all of the conversations I had in my head as I was putting all of this together and I kept going in circles as to what I felt was best, so I just picked one and went with it.  I think all of the approaches have fairly equivalent pros and cons, so I'm not opposed to any of them.

The reason I landed on the implementation here was because in my head I was seeing this list as an "override" list, which would mean it would override the `sign_up_allowed` setting in any state.

I'll do whatever everyone thinks is best though.

Reply:
I think the best would be either to just have one config option or not to have them override each other, otherwise, it's easy to make a mistake without reading the docs. @jamilbk wdyt?

Reply:
No strong opinion. I don't think this feature will be user-facing, right? Signups are disabled on staging and will be open on prod, so whatever fits that use case best works for me.

Reply:
I've had this running in the back of my mind all day and I think I'd like to stick with the existing implementation with a slight change.  My reasoning is, after thinking through all of the options, they all have equal downsides in my opinion and would require reading the docs to understand the behavior, so let's make as few changes as possible.

The one change I think I'd like to make is to change the name of the config variable to `sign_up_always_allowed_domains`.  I'm hopeful that it would be fairly easy to reason about what the behavior would be...You can turn the sign up feature on or off, but if there's something in the "always allow" list then those things are always allowed.

With the other options, each of them has an "allow list", but an empty list means allow all, which I feel isn't necessary intuitive and would require reading some form of documentation to be sure of what the behavior is.  When I think of "allow lists" I jump to "this list is what is allowed" and if it's empty I jump to "nothing is allowed". 

@AndrewDryga, are you strongly opposed to this?

Reply:
@bmanifold I just already had a "config migrations" trauma with 0.7 so would really like to have it cleaned out. I know we want this feature "just for us" but it's open source and there are people already self-hosting without any docs at all, don't want to let things that are easy to fix now to be growing to be a bigger problem later (even just having something we need to keep in mind ourselves).

Reply:
@AndrewDryga Alright, in that case, let's try and sync up and figure out a solution then.

Reply:
@bmanifold sorry for making you do it, let's try to keep it as simple as possible: one flag turns sign ups one/off completely and second controls the whitelist. I believe this will require the least amount of conditional code.